lib/puppet/node.rb

```diff
@@ -45,10 +45,20 @@ def to_data_hash
       'environment' => environment.name.to_s,
     }
     result['classes'] = classes unless classes.empty?
-    result['parameters'] = parameters unless parameters.empty?
+    result['parameters'] = serializable_parameters unless serializable_parameters.empty?
     result
   end
 
+  def to_json
+    self.to_data_hash.to_json
+  end
+
+  def serializable_parameters
+    new_params = parameters.dup
```

Comment:
should this be '@parameters'? (It essential is because of attr_accessor :parameters, but it would be nice if this was consistent I guess)